lib/airbrake-ruby/async_sender.rb

```diff
@@ -24,6 +24,8 @@ def initialize(config)
     #   library
     # @return [nil]
     def send(notice)
+      return will_not_deliver(notice) if @unsent.size >= @unsent.max
```

Comment:
Agree, but at the same time it doesn't hurt and, since there are threads around, It is more robust and cant protect against some weird race condition that could manage to increment the size value over the max. If that could happen somehow, this condition would minimise the harm (since it will still prevent from more elements being pushed into the queue).